### PR TITLE
EPMRPP-37317 Admin. Projects. Assign user on redirect to project.

### PR DESCRIPTION
**Pull request by ElenaRomanchuk**
EPMRPP-37317. Admin. Projects. Assign user on redirect to project.
! Based on EPMRPP-37395. Please, check and merge PR https://github.com/reportportal/service-ui/pull/1310 at first.

### CODE REVIEW

File: app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+    id: 'ProjectsPage.modal.deleteModalConfirmationText',
+    defaultMessage: 'Are you sure you want to delete the project <b>{name}</b>?',
+  },
+  deleteModalSubmitButtonText: {
+    id: 'ProjectsPage.modal.deleteModalSubmitButtonText',
+    defaultMessage: 'Delete',
+  },
+  deleteError: {
+    id: 'ProjectsPage.deleteError',
+    defaultMessage: 'An error occurred during deleting the project',
```
Comment (margalencia) [inline app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: I have the same messages for the bulk `delete` operation in Grid view in `projectsPage/projectsToolbar/projectsToolbar.jsx`, maybe we can use a shared mesage config for them?

Comment (ElenaRomanchuk) [reply app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: Is it a good idea to extract all messages into separate file for common use in Projects?

Comment (DiscoElevator) [reply app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: I think it's a good idea

Comment (ElenaRomanchuk) [reply app/src/pages/admin/projectsPage/projectMenu/projectMenu.jsx]: Done!

File: app/src/pages/common/modals/confirmModal.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    return (
+      <ModalLayout
+        title={title}
+        okButton={{
+          text: confirmText,
+          onClick: (closeModal) => {
+            confirmModal();
+            closeModal();
+            onConfirm();
+            hideModal();
```
Comment (DiscoElevator) [inline app/src/pages/common/modals/confirmModal.jsx]: Is it necessary to use `hideModal` after `closeModal`? As far as I see it is called automatically after closing a modal.

File: app/src/pages/common/modals/confirmModal.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import { connect } from 'react-redux';
+import { ModalLayout, withModal } from 'components/main/modal';
+import { hideModalAction, confirmModalAction } from 'controllers/modal';
+
+@withModal('confirmModal')
+@connect(null, {
+  confirmModal: confirmModalAction,
+  hideModal: hideModalAction,
+})
+export class ConfirmModal extends Component {
```
Comment (DiscoElevator) [inline app/src/pages/common/modals/confirmModal.jsx]: I think `ConfirmationModal` sounds better (so no confusion between `confirmModal` as an action and `ConfirmModal` as a component)

File: app/src/pages/common/modals/confirmModal.jsx (hunk 2 above)
Comment (ElenaRomanchuk) [reply app/src/pages/common/modals/confirmModal.jsx]: `hideModal` is called in `onExited` event of `CSSTransition` component in `modalLayout`. Unfortunately, it doesn't work properly when you perform an asynchronous redirect to another page.

Comment (DiscoElevator) [reply app/src/pages/common/modals/confirmModal.jsx]: Maybe dispatching `hideModalAction` in saga would look less confusing.

Comment (ElenaRomanchuk) [reply app/src/pages/common/modals/confirmModal.jsx]: I suggest dispatch it in confirmModalAction. Would it be OK?

Comment (DiscoElevator) [reply app/src/pages/common/modals/confirmModal.jsx]: You can't dispatch two actions in one (there's a task to remove all thunk usage in favor of saga - no need to keep both libraries).
Somewhere [here](https://github.com/reportportal/service-ui/pull/1317/files#diff-4b647578726008404857fb737592a00cR11), I think:
```
if (confirmed) {
  yield put(hideModalAction());
}
```

**Pull request merged**